### PR TITLE
Remove Goog math

### PR DESCRIPTION
**Pull request by nicholas-l**
Work in progress to remove goog.math.

### CODE REVIEW

**Comment by tschaub** [comment]
@nicholas-l - any updates on this?

File: src/ol/reproj/reproj.js [hunk 1]
```diff
@@ -61,7 +60,7 @@ ol.reproj.calculateSourceResolution = function(sourceProj, targetProj,
       sourceProj.getPointResolution(sourceResolution, sourceCenter) /
       sourceResolution;
 
-  if (goog.math.isFiniteNumber(compensationFactor) && compensationFactor > 0) {
+  if (isFinite(compensationFactor) && !isNaN(compensationFactor) && compensationFactor > 0) {
```
Comment (tschaub) [inline src/ol/reproj/reproj.js]: This was changed with #4884.

File: src/ol/reproj/tile.js [hunk 2]
```diff
@@ -138,7 +137,7 @@ ol.reproj.Tile = function(sourceProj, sourceTileGrid,
   var sourceResolution = ol.reproj.calculateSourceResolution(
       sourceProj, targetProj, targetCenter, targetResolution);
 
-  if (!goog.math.isFiniteNumber(sourceResolution) || sourceResolution <= 0) {
+  if (!(isFinite(sourceResolution) && !isNaN(sourceResolution)) || sourceResolution <= 0) {
```
Comment (tschaub) [inline src/ol/reproj/tile.js]: Also handled in #4884.

**Comment by nicholas-l** [comment]
No sorry, so anyone can grab it and run

**Comment by nicholas-l** [comment]
@tschaub Rebased and finished the rest of goog.math.*. Ready for review

File: src/ol/dom/dom.js [hunk 3]
```diff
@@ -213,3 +213,20 @@ ol.dom.outerHeight = function(element) {
 
   return height;
 };
+
+ol.dom.setBoxSizingSize = function(element, size, boxSizing) {
```
Comment (tschaub) [inline src/ol/dom/dom.js]: It looks like this is unused.

**Comment by tschaub** [comment]
@nicholas-l it looks like `ol.dom.setBoxSizingSize()` can be removed.  And it would be great to have tests for `ol.math.lerp()` as well.

**Comment by tschaub** [comment]
If you rebase and edit `077358e1c3738e0f7fe6ec15212effb88e68e2be`, you can remove `ol.dom.setBoxSizingSize()` without an additional commit.

File: test/spec/ol/math.test.js [hunk 4]
```diff
@@ -155,5 +155,29 @@ describe('ol.math.toRadians', function() {
...
+    expect(ol.math.modulo(7, 8)).to.be(7);
+  });
+  it('same Dividend and Divisor returns 0', function() {
+    expect(ol.math.modulo(4, 4)).to.be(0);
+  });
+  it('negative and positive returns negative', function() {
+    expect(ol.math.modulo(-3, 4)).to.be(1);
+  });
+  it('negative and negative returns positive', function() {
+    expect(ol.math.modulo(-4, -5)).to.be(-4);
```
Comment (marcjansen) [inline test/spec/ol/math.test.js]: Either the  message is wrong (returns positive) or the method.

File: test/spec/ol/math.test.js [hunk 5]
```diff
@@ -155,5 +155,29 @@ describe('ol.math.toRadians', function() {
...
+    expect(ol.math.modulo(256, 8)).to.be(0);
+  });
+  it('positive and positive returns a positive ', function() {
+    expect(ol.math.modulo(7, 8)).to.be(7);
+  });
+  it('same Dividend and Divisor returns 0', function() {
+    expect(ol.math.modulo(4, 4)).to.be(0);
+  });
+  it('negative and positive returns negative', function() {
+    expect(ol.math.modulo(-3, 4)).to.be(1);
```
Comment (marcjansen) [inline test/spec/ol/math.test.js]: Either the message is wrong (returns negative) or the method.

**Comment by nicholas-l** [comment]
@tschaub @marcjansen Updated according to your comments and added tests for lerp

**Comment by gberaudo** [comment]
+1. A good candidate for squashing?

**Pull request merged**